### PR TITLE
[repo] Include license files in packages

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #5187

## Changes

* Include license files in packages.

* Rename `LICENSE` -> `LICENSE.TXT` and `THIRD-PARTY-NOTICES` -> `THIRD-PARTY-NOTICES.TXT`. Without a file extension when packaged they end up going into their own folders. With a file extension they end up in the root: 

   image

## Details

For this work I am using dotnet/runtime as guide:

* https://github.com/dotnet/runtime/blob/8bb7b2ac092a6580b1cc247bc705e737c2b0260d/Directory.Build.props#L357
* https://github.com/dotnet/runtime/blob/8bb7b2ac092a6580b1cc247bc705e737c2b0260d/Directory.Build.targets#L62-L71

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: build/Common.prod.props
Comment (reyang) [inline build/Common.prod.props:24]: Do we include this in all packages or only the packages that actually have 3rd party code?

Comment (CodeBlanch) [reply build/Common.prod.props:24]: The current design is to include it in all packages. Overkill, but much simpler that way! dotnet\runtime does the same. For example here is `System.Diagnostics.DiagnosticSource`:

https://nuget.info/packages/System.Diagnostics.DiagnosticSource/8.0.0

It has the main repo `THIRD-PARTY-NOTICES.txt` which includes license notices for things like ASP.NET, mono, Zlib, etc. Doubt any of that is really in DS.

Comment (reyang) [reply build/Common.prod.props:24]: Got it, thanks for the context 👍

File: build/Common.prod.props
Comment (reyang) [inline build/Common.prod.props:22]: Do we want this? .NET seems to use `false`. https://github.com/dotnet/runtime/blob/8bb7b2ac092a6580b1cc247bc705e737c2b0260d/Directory.Build.props#L355

Comment (CodeBlanch) [reply build/Common.prod.props:22]: I noticed that too. Not sure why we have `true` there TBH. I'll bring up on the next SIG for discussion.

Comment (reyang) [reply build/Common.prod.props:22]: Not a blocker for this PR.

File: LICENSE.txt
Comment (utpilla) [inline LICENSE.txt:1]: It looks like GitHub UI does some special rendering for the license file. That does not seem to be the case when I open the license file on your repo's feature branch.: https://github.com/CodeBlanch/opentelemetry-dotnet/tree/package-license-files Do you know why would that be the case?

This is how it looks when you open the license file on the main repo:
image

File: OpenTelemetry.sln
Comment (CodeBlanch) [inline OpenTelemetry.sln:19]:
```suggestion
		LICENSE.TXT = LICENSE.TXT
```

File: OpenTelemetry.sln
Comment (CodeBlanch) [inline OpenTelemetry.sln:23]:
```suggestion
		THIRD-PARTY-NOTICES.TXT = THIRD-PARTY-NOTICES.TXT
```

File: LICENSE.txt
Comment (CodeBlanch) [reply LICENSE.txt:1]: @utpilla Good question! I renamed the files `.txt` -> `.TXT` so they matched runtime but it didn't change the rendering. ~Googled~ Binged around a bit but couldn't find any good doc pointing to known extensions or names or anything which might explain it.

I notice if I browse the root of my [branch](https://github.com/CodeBlanch/opentelemetry-dotnet/tree/package-license-files) it still shows the license from the main branch:

image

My guess is this is a GitHub quirk and once we merge the PR it will work fine. How about we merge this and see what happens? If it doesn't work I'll open a follow-up to do _something_ 🤣

Comment (utpilla) [reply LICENSE.txt:1]: I think when you simply rename the file, GitHub might be losing the template related information.

You could probably try, deleting the existing file and then create a new file named LICENSE.txt using the GitHub UI where  it would let you select the template:

image

**Review by utpilla — APPROVED** [review]

File: LICENSE.txt
Comment (utpilla) [reply LICENSE.txt:1]: > My guess is this is a GitHub quirk and once we merge the PR it will work fine. How about we merge this and see what happens? If it doesn't work I'll open a follow-up to do something 🤣

We could do that.

Comment (utpilla) [reply LICENSE.txt:1]: Just merged the PR and it works.

Comment (CodeBlanch) [reply LICENSE.txt:1]: Sweet! Thanks for confirming.

**Pull request merged**
